packages/amplify-e2e-tests/src/__tests__/resolvers.test.ts

```diff
@@ -47,6 +47,30 @@ describe('user created resolvers', () => {
 
       expect(fs.readFileSync(generatedResolverPath).toString()).toEqual(resolver);
     });
+
+    it('overriding a resolver should not create duplicate function', async () => {
+      const slotName = 'Query.listTodos.auth.1.req.vtl';
+      const slot = '$util.unauthorized()';
+      const generatedResolverPath = join(projectDir, 'amplify', 'backend', 'api', apiName, 'build', 'resolvers', slotName);
+
+      await addApiWithoutSchema(projectDir, { apiName });
+      updateApiSchema(projectDir, apiName, 'cognito_simple_model.graphql');
+      await amplifyPushGraphQlWithCognitoPrompt(projectDir);
+
+      expect(fs.existsSync(generatedResolverPath)).toEqual(true);
+
+      addCustomResolver(projectDir, apiName, slotName, slot);
+      await generateModels(projectDir);
+      await amplifyPushUpdate(projectDir);
+
+      const todoJsonPath = join(projectDir, 'amplify', 'backend', 'api', apiName, 'build', 'stacks', 'Todo.json');
+      const todoJson = JSON.parse(fs.readFileSync(todoJsonPath).toString());
+
+      expect(fs.readFileSync(generatedResolverPath).toString()).toEqual(slot);
+      expect(todoJson.Resources.GetTodoResolver.Properties.PipelineConfig.Functions).toHaveLength(3);
+      // The function count should be 3 even after overriding the auth resolver
+      expect(todoJson.Resources.ListTodoResolver.Properties.PipelineConfig.Functions).toHaveLength(3);
```

Comment:
If it's not too much additional work, can we add a check that the contents of the custom resolver slot are preserved and not overwritten? Thanks

Reply:
Updated the e2e test to check the overridden resolver folder and make sure that the file exists and the contents match the custom resolver content.